tests/client/amo/sagas/testCategories.js

```diff
@@ -0,0 +1,119 @@
+import { hideLoading, showLoading } from 'react-redux-loading-bar';
+import { takeEvery } from 'redux-saga/effects';
+import SagaTester from 'redux-saga-tester';
+
+import categoriesSaga, { fetchCategories } from 'amo/sagas/categories';
+import createStore from 'amo/store';
+import * as actions from 'core/actions/categories';
+import * as api from 'core/api';
+import {
+  CATEGORIES_FAIL,
+  CATEGORIES_FETCH,
+  CATEGORIES_LOAD,
+} from 'core/constants';
+import apiReducer from 'core/reducers/api';
+import categoriesReducer from 'core/reducers/categories';
+import { signedInApiState } from 'tests/client/amo/helpers';
+
+
+describe('categoriesSaga', () => {
+  let initialState;
+  let sagaTester;
+  let store;
+
+  beforeEach(() => {
+    store = createStore().store;
+    const state = store.getState();
+    initialState = {
+      api: signedInApiState,
+      categories: state.categories,
+    };
+
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: { api: apiReducer, categories: categoriesReducer },
+    });
+    sagaTester.start(categoriesSaga);
+  });
+
+  it('should get Api from state then make API request to categories', async () => {
+    const mockApi = sinon.mock(api);
+    const entities = sinon.stub();
+    const result = sinon.stub();
+
+    mockApi
+      .expects('categories')
+      .once()
+      .withArgs({
+        api: { ...initialState.api },
+      })
+      .returns(Promise.resolve({ entities, result }));
+
+    assert.deepEqual(sagaTester.getState(), initialState);
+
+    sagaTester.dispatch(actions.categoriesFetch());
+
+    assert.deepEqual(sagaTester.getState(), {
+      ...initialState,
+      categories: { ...initialState.categories, loading: true },
+    });
+
+    await sagaTester.waitFor(CATEGORIES_LOAD);
+
+    const calledActions = sagaTester.getCalledActions();
+
+    // First action is CATEGORIES_FETCH.
+    assert.deepEqual(calledActions[0], actions.categoriesFetch());
+
+    // Next action is showing the loading bar.
+    assert.deepEqual(calledActions[1], showLoading());
+
+    // Next action is loading the categories returned by the API.
+    assert.deepEqual(calledActions[2], actions.categoriesLoad({ result }));
+
+    // Last action is to hide the loading bar.
+    assert.deepEqual(calledActions[3], hideLoading());
+
+    mockApi.verify();
+  });
+
+  it('should dispatch fail if API request fails', async () => {
+    const mockApi = sinon.mock(api);
+    const error = sinon.stub();
+
+    mockApi
+      .expects('categories')
+      .once()
+      .withArgs({
+        api: { ...initialState.api },
+      })
+      .throws(error);
```

Comment:
`mockApi.throws(error);` seems to cause an error because `mockApi.throws is not a function`. I can trim it down to `mockApi.expects('categories').throws(error);`